app/Model/Event.php

```diff
@@ -6848,4 +6848,27 @@ public function addFiltersFromSubqueryElements($filters, $subqueryElements)
         }
         return $filters;
     }
+
+    /**
+     * @param array $event
+     */
+    public function removeGalaxyClusterTags(array &$event)
```

Comment:
Simple question/suggestion. Why not use `isset()` comparison on associative arrays instead of `in_array()`?
Something like this:
```
$galaxyTagIds = array();
foreach ($event['Galaxy'] as $galaxy) {
    foreach ($galaxy['GalaxyCluster'] as $galaxyCluster) {
        $galaxyTagIds[$galaxyCluster['tag_id']] = true;
    }
}

if (empty($galaxyTagIds)) {
    return;
}
foreach ($event['EventTag'] as $k => $eventTag) {
    if (isset($galaxyTagIds[$eventTag['tag_id']])) {
        unset($event['EventTag'][$k]);
    }
}
```

Reply:
Changed, I was thinking that it should be not necessary, because it will contains just few tags,  but still make sense.